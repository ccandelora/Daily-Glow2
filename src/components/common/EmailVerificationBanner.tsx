import React from 'react';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';
import { Ionicons } from '@expo/vector-icons';

interface EmailVerificationBannerProps {
  email: string;
  onResendVerification?: () => void;
}

export const EmailVerificationBanner = ({ 
  email, 
  onResendVerification 
}: EmailVerificationBannerProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <Ionicons name="mail-unread-outline" size={24} color={theme.COLORS.ui.text} />
      </View>
      <View style={styles.textContainer}>
        <Typography variant="body" style={styles.title}>
          Please verify your email
        </Typography>
        <Typography variant="caption" style={styles.message}>
          We've sent a verification link to {email}. Please check your inbox and verify your email to unlock all features.
        </Typography>
        {onResendVerification && (
          <TouchableOpacity onPress={onResendVerification} style={styles.resendButton}>
            <Typography variant="caption" style={styles.resendText}>
              Resend verification email
            </Typography>
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'rgba(255, 193, 7, 0.15)',
    borderRadius: theme.SPACING.md,
    padding: theme.SPACING.md,
    flexDirection: 'row',
    marginVertical: theme.SPACING.md,
    borderWidth: 1,
    borderColor: 'rgba(255, 193, 7, 0.3)',
  },
  iconContainer: {
    marginRight: theme.SPACING.md,
    alignItems: 'center',
    justifyContent: 'center',
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: 'rgba(255, 193, 7, 0.2)',
  },
  textContainer: {
    flex: 1,
  },
  title: {
    fontWeight: theme.FONTS.weights.bold as any,
    marginBottom: theme.SPACING.xs,
    color: theme.COLORS.ui.text,
  },
  message: {
    color: theme.COLORS.ui.textSecondary,
    lineHeight: 20,
  },
  resendButton: {
    marginTop: theme.SPACING.sm,
  },
  resendText: {
    color: theme.COLORS.primary.green,
    textDecorationLine: 'underline',
    fontWeight: theme.FONTS.weights.bold as any,
  },
}); 