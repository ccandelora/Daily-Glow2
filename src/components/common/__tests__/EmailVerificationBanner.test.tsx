import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { EmailVerificationBanner } from '../EmailVerificationBanner';

// Mock the Ionicons component
jest.mock('@expo/vector-icons', () => ({
  Ionicons: 'Ionicons',
}));

describe('EmailVerificationBanner', () => {
  const testEmail = 'test@example.com';
  
  it('renders correctly with email', () => {
    const { toJSON, getByText } = render(
      <EmailVerificationBanner email={testEmail} />
    );
    
    expect(getByText('Please verify your email')).toBeTruthy();
    expect(getByText(`We've sent a verification link to ${testEmail}. Please check your inbox and verify your email to unlock all features.`)).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('does not show resend button when onResendVerification is not provided', () => {
    const { queryByText } = render(
      <EmailVerificationBanner email={testEmail} />
    );
    
    expect(queryByText('Resend verification email')).toBeNull();
  });

  it('shows resend button when onResendVerification is provided', () => {
    const { getByText } = render(
      <EmailVerificationBanner 
        email={testEmail} 
        onResendVerification={() => {}}
      />
    );
    
    expect(getByText('Resend verification email')).toBeTruthy();
  });

  it('calls onResendVerification when resend button is pressed', () => {
    const mockResend = jest.fn();
    const { getByText } = render(
      <EmailVerificationBanner 
        email={testEmail} 
        onResendVerification={mockResend}
      />
    );
    
    fireEvent.press(getByText('Resend verification email'));
    expect(mockResend).toHaveBeenCalledTimes(1);
  });
}); 