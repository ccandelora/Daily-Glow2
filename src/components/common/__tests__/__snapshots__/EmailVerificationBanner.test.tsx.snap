// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailVerificationBanner renders correctly with email 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255, 193, 7, 0.15)",
      "borderColor": "rgba(255, 193, 7, 0.3)",
      "borderRadius": 16,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginVertical": 16,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 193, 7, 0.2)",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 16,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#FFFFFF"
      name="mail-unread-outline"
      size={24}
    />
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          undefined,
          null,
          {
            "color": "#FFFFFF",
            "fontWeight": "700",
            "marginBottom": 4,
          },
        ]
      }
    >
      Please verify your email
    </Text>
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 14,
            "fontWeight": "400",
          },
          undefined,
          null,
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "lineHeight": 20,
          },
        ]
      }
    >
      We've sent a verification link to 
      test@example.com
      . Please check your inbox and verify your email to unlock all features.
    </Text>
  </View>
</View>
`;
